Sources/Identity/CustomerInfoManager.swift

```diff
@@ -319,20 +319,28 @@ private extension CustomerInfoManager {
                          completion: @escaping CustomerAPI.CustomerInfoResponseHandler) {
         if #available(iOS 15.0, tvOS 15.0, macOS 12.0, watchOS 8.0, *) {
             _ = Task<Void, Never> {
-                // Note: this is only able to post a single transaction,
-                // it can be improved in the future once `PostReceiptOperation` accepts multiple ones.
-                if let transaction = await self.transactionFetcher.unfinishedVerifiedTransactions.first {
-                    Logger.debug(Strings.customerInfo.posting_transaction_in_lieu_of_fetching_customerinfo(transaction))
-
-                    self.transactionPoster.handlePurchasedTransaction(
-                        transaction,
-                        data: .init(appUserID: appUserID,
-                                    presentedOfferingID: nil,
-                                    unsyncedAttributes: [:],
-                                    storefront: await Storefront.currentStorefront,
-                                    source: Self.sourceForUnfinishedTransaction),
-                        completion: completion
+                let transactions = await self.transactionFetcher.unfinishedVerifiedTransactions
+
+                if !transactions.isEmpty {
+                    var lastResult: Result<CustomerInfo, BackendError> = .failure(.missingCachedCustomerInfo())
+                    let storefront = await Storefront.currentStorefront
+
+                    Logger.debug(
+                        Strings.customerInfo.posting_transactions_in_lieu_of_fetching_customerinfo(transactions)
                     )
+
+                    for transaction in transactions {
+                        lastResult = await self.transactionPoster.handlePurchasedTransaction(
```

Comment:
Hmm I think this way we would override the previous result even if it was an error right? So if the last request succeeded but any of the previous ones failed, this would return success.
Should we post the first error if any? So the logic I think would be best would be:
- If any post fails, keep posting but keep track of the errors.
- Once all post end, if there were no errors, continue with the last success, or with the first error if any.

Reply:
> Once all post end, if there were no errors, continue with the last success, or with the first error if any.

Considering that every post will post the entirety of the receipt, I think it would be best to be more optimistic with the results. This comes from calling `getCustomerInfo`. If we could post any of these, and get a `CustomerInfo`, it would be up-to-date.

I'll update the logic to pick any of the successes instead of "randomly" the last one.

Reply:
Ahhh right, it's different in iOS true. Then yeah, that makes sense 👍

Reply:
Done: f9ed157599c60f8912e3a4fa684270d33e9acd42